ExchangeSharp/API/Exchanges/Digifnex/ExchangeDigifinexAPI.cs

```diff
@@ -0,0 +1,514 @@
+            var list = token["data"];
+            return list.Select(x => new ExchangeOrderResult
+            {
+                MarketSymbol = (string)x["symbol"],
+                OrderId = (string)x["order_id"],
+                OrderDate = DateTimeOffset.FromUnixTimeSeconds((long)x["created_date"]).LocalDateTime,
+                FillDate = DateTimeOffset.FromUnixTimeSeconds((long)x["finished_date"]).LocalDateTime,
+                Price = (decimal)x["price"],
+                AveragePrice = (decimal)x["avg_price"],
+                Amount = (decimal)x["amount"],
+                AmountFilled = (decimal)x["executed_amount"],
+                IsBuy = (string)x["type"] == "buy",
+                Result = ParseOrderStatus(x["status"]),
+            });
+        }
+
+        protected override async Task<IEnumerable<ExchangeOrderResult>> OnGetCompletedOrderDetailsAsync(
+            string marketSymbol = null, DateTime? afterDate = null)
+        {
+            Dictionary<string, object> payload = await GetNoncePayloadAsync();
+            var url = "/spot/mytrades?limit=500";
+
+            if (marketSymbol?.Length > 0)
+                url += "&symbol=" + marketSymbol;
+
+            if (afterDate != null)
+            {
+                var startTime = (long)afterDate.Value.UnixTimestampFromDateTimeSeconds();
+                url += "&start_time=" + startTime.ToString();
+            }
+
+            JToken token = await MakeJsonRequestAsync<JToken>(url, payload: payload);
+            var list = token["list"];
+            return list.Select(x => new ExchangeOrderResult
+            {
+                MarketSymbol = (string)x["symbol"],
+                OrderId = (string)x["order_id"],
+                TradeId = (string)x["id"],
+                Price = (decimal)x["price"],
+                AmountFilled = (decimal)x["amount"],
+                Fees = (decimal)x["fee"],
+                FeesCurrency = (string)x["fee_currency"],
+                FillDate = DateTimeOffset.FromUnixTimeSeconds((long)x["timestamp"]).LocalDateTime,
+                IsBuy = (string)x["side"] == "buy",
+                Result = ExchangeAPIOrderResult.Unknown,
+            });
+        }
+
+        protected override async Task<ExchangeOrderResult> OnGetOrderDetailsAsync(string orderId, string marketSymbol = null)
+        {
+            Dictionary<string, object> payload = await GetNoncePayloadAsync();
+            JToken token = await MakeJsonRequestAsync<JToken>($"/spot/order?order_id={orderId}", payload: payload);
+            var x = token["data"];
+            return new ExchangeOrderResult
+            {
+                MarketSymbol = (string)x["symbol"],
+                OrderId = (string)x["order_id"],
+                OrderDate = DateTimeOffset.FromUnixTimeSeconds((long)x["created_date"]).LocalDateTime,
+                FillDate = DateTimeOffset.FromUnixTimeSeconds((long)x["finished_date"]).LocalDateTime,
```

Comment:
There is a method CryptoUtility.UnixTimeStampToDateTimeSeconds() that can help you do this conversion but keep the timestamp in UTC.

Reply:
Thanks so much for changing to utc, can you switch all the (long) and (string) casts to use ConvertInvariant<long>() and ToStringInvariant()? This is the standard way to convert longs and strings to avoid any chance of culture parsing and formating issues. We are close now!